tests/Unit/RuleSet/RuleSetTest.php

```diff
@@ -305,6 +305,46 @@ public function addRuleWithSiblingNotInRuleSetAddsRuleAfterInitialRulesAndSetsVa
         self::assertGreaterThanOrEqual(0, $ruleToAdd->getColumnNumber(), 'column number not valid');
     }
 
+    /**
+     * @test
+     *
+     * @param non-empty-list<string> $initialPropertyNames
+     * @param int<0, max> $indexToRemove
+     *
+     * @dataProvider provideInitialPropertyNamesAndIndexOfOne
+     */
+    public function removeRuleRemovesRuleInSetAndKeepsOthers(array $initialPropertyNames, int $indexToRemove): void
+    {
+        $this->setRulesFromPropertyNames($initialPropertyNames);
+        $ruleToRemove = $this->subject->getRules()[$indexToRemove];
+
+        $this->subject->removeRule($ruleToRemove);
+
+        $remainingRules = $this->subject->getRules();
+        self::assertNotContains($ruleToRemove, $remainingRules);
+        self::assertCount(\count($initialPropertyNames) - 1, $remainingRules);
+    }
+
+    /**
+     * @test
+     *
+     * @param list<string> $initialPropertyNames
+     *
+     * @dataProvider provideInitialPropertyNamesAndAnotherPropertyName
+     */
+    public function removeRuleDoesNothingWithRuleNotInSet(
```

Comment:
Let's reword this to describe what is happening instead of what isn't happening:

```suggestion
    public function removeRuleWithRuleNotInSetKeepsSetUnchanged
```